src/routers/StackRouter.js

```diff
@@ -186,7 +186,11 @@ export default (
       }
 
       // Handle navigation to other child routers that are not yet pushed
-      if (action.type === NavigationActions.NAVIGATE) {
+      if (
+        action.type === NavigationActions.NAVIGATE &&
+        action.routeName !== 'DrawerOpen' &&
```

Comment:
Can you recommend a different way to do this?  My understanding is that this code manages navigation to child routers that are not yet pushed but this approach seems to be creating a bug where existing child drawer navigators are being executed instead of the parent drawer navigator.  This was not a problem before this piece of code was introduced in https://github.com/react-community/react-navigation/commit/2dc869fedd27d304c2f0ddd606761320b8bec923.  Am I missing something or is there a better way to do this?